x-pack/plugins/ml/public/application/explorer/anomaly_detection_alerts_state_service.ts

```diff
@@ -0,0 +1,174 @@
+    private readonly timefilter: TimefilterContract
+  ) {
+    super();
+    this._init();
+
+    this.selectedAlerts$ = combineLatest([
+      this._aadAlerts$,
+      this._anomalyTimelineStateServices.getSelectedCells$().pipe(map((cells) => cells?.times)),
+    ]).pipe(
+      map(([alerts, selectedTimes]) => {
+        if (!Array.isArray(selectedTimes)) return null;
+
+        return alerts.filter(
+          (alert) =>
+            alert.anomalyTimestamp >= selectedTimes[0] * 1000 &&
+            alert.anomalyTimestamp <= selectedTimes[1] * 1000
+        );
+      })
+    );
+  }
+
+  public readonly anomalyDetectionAlerts$: Observable<AnomalyDetectionAlert[]> =
+    this._aadAlerts$.asObservable();
+
+  public readonly selectedAlerts$: Observable<AnomalyDetectionAlert[] | null>;
+
+  public readonly countByStatus$: Observable<Record<string, number>> = this._aadAlerts$.pipe(
+    map((alerts) => {
+      return alerts.reduce(
+        (acc, alert) => {
+          if (!isDefined(acc[alert.alertStatus])) {
+            acc[alert.alertStatus] = 0;
+          } else {
+            acc[alert.alertStatus]++;
+          }
+          return acc;
+        },
+        { active: 0, recovered: 0 } as Record<string, number>
+      );
+    })
+  );
+
+  protected _initSubscriptions(): Subscription {
+    const subscription = new Subscription();
+
+    const timeBounds$ = this.timefilter.getTimeUpdate$().pipe(
+      startWith(null),
+      map(() => this.timefilter.getBounds())
+    );
+
+    subscription.add(
+      combineLatest([this._anomalyTimelineStateServices.getSwimLaneJobs$(), timeBounds$])
+        .pipe(
+          // Create a result query from the input
+          map(([selectedJobs, timeBounds]) => {
+            return {
+              bool: {
+                must: [
+                  {
+                    term: {
+                      [ALERT_RULE_TYPE_ID]: ML_ALERT_TYPES.ANOMALY_DETECTION,
+                    },
+                  },
+                  {
+                    range: {
+                      [ALERT_ANOMALY_TIMESTAMP]: {
+                        gte: timeBounds.min?.valueOf(),
+                        lte: timeBounds.max?.valueOf(),
+                      },
+                    },
+                  },
+                  {
+                    terms: {
+                      [ALERT_ANOMALY_DETECTION_JOB_ID]: selectedJobs.map((job) => job.id),
+                    },
+                  },
+                ],
+              },
+            } as RuleRegistrySearchRequest['query'];
+          }),
+          switchMap((query) => {
+            return this.data.search.search<RuleRegistrySearchRequest, RuleRegistrySearchResponse>(
+              {
+                featureIds: [AlertConsumers.ML],
+                query,
+              },
+              { strategy: 'privateRuleRegistryAlertsSearchStrategy' }
+            );
+          })
+        )
+        .subscribe((response) => {
```

Comment:
What would happen if there's an error with response?

Reply:
Added error catching in https://github.com/elastic/kibana/pull/167998/commits/fee24b741b6a6e5235146a517582bb0ce62b866e